src/data/java/com/github/minecraftschurli/arsmagicalegacy/data/AMEnglishLanguageProvider.java

```diff
@@ -0,0 +1,196 @@
+        blockIdTranslation(AMBlocks.WITCHWOOD_PLANKS);
+        blockIdTranslation(AMBlocks.WITCHWOOD_SLAB);
+        blockIdTranslation(AMBlocks.WITCHWOOD_STAIRS);
+        blockIdTranslation(AMBlocks.WITCHWOOD_FENCE);
+        blockIdTranslation(AMBlocks.WITCHWOOD_FENCE_GATE);
+        blockIdTranslation(AMBlocks.WITCHWOOD_DOOR);
+        blockIdTranslation(AMBlocks.WITCHWOOD_TRAPDOOR);
+        blockIdTranslation(AMBlocks.WITCHWOOD_BUTTON);
+        blockIdTranslation(AMBlocks.WITCHWOOD_PRESSURE_PLATE);
+        itemIdTranslation(AMItems.BLANK_RUNE);
+        itemIdTranslation(AMItems.WHITE_RUNE);
+        itemIdTranslation(AMItems.ORANGE_RUNE);
+        itemIdTranslation(AMItems.MAGENTA_RUNE);
+        itemIdTranslation(AMItems.LIGHT_BLUE_RUNE);
+        itemIdTranslation(AMItems.YELLOW_RUNE);
+        itemIdTranslation(AMItems.LIME_RUNE);
+        itemIdTranslation(AMItems.PINK_RUNE);
+        itemIdTranslation(AMItems.GRAY_RUNE);
+        itemIdTranslation(AMItems.LIGHT_GRAY_RUNE);
+        itemIdTranslation(AMItems.CYAN_RUNE);
+        itemIdTranslation(AMItems.PURPLE_RUNE);
+        itemIdTranslation(AMItems.BLUE_RUNE);
+        itemIdTranslation(AMItems.BROWN_RUNE);
+        itemIdTranslation(AMItems.GREEN_RUNE);
+        itemIdTranslation(AMItems.RED_RUNE);
+        itemIdTranslation(AMItems.BLACK_RUNE);
+        itemIdTranslation(AMItems.RUNE_BAG);
+        itemIdTranslation(AMItems.ARCANE_COMPOUND);
+        itemIdTranslation(AMItems.ARCANE_ASH);
+        itemIdTranslation(AMItems.PURIFIED_VINTEUM_DUST);
+        blockIdTranslation(AMBlocks.AUM);
+        blockIdTranslation(AMBlocks.CERUBLOSSOM);
+        blockIdTranslation(AMBlocks.DESERT_NOVA);
+        blockIdTranslation(AMBlocks.TARMA_ROOT);
+        blockIdTranslation(AMBlocks.WAKEBLOOM);
+        blockIdTranslation(AMBlocks.VINTEUM_TORCH);
+        blockTranslation(AMBlocks.WIZARDS_CHALK, "Wizard's Chalk");
+        advancementTranslation("root", ArsMagicaLegacy.getModName(), "A renewed look into Minecraft with a splash of magic...");
+    }
+
+    /**
+     * Adds a block translation that matches the block id.
+     *
+     * @param block The block to generate the translation for.
+     */
+    private void blockIdTranslation(Supplier<? extends Block> block) {
+        addBlock(block, idToTranslation(block.get().getRegistryName().getPath()));
+    }
+
+    /**
+     * Adds a block translation.
+     *
+     * @param block       The block to generate the translation for.
+     * @param translation The translation to apply.
+     */
+    private void blockTranslation(Supplier<? extends Block> block, String translation) {
+        addBlock(block, translation);
+    }
+
+    /**
+     * Adds an item translation that matches the item id.
+     *
+     * @param item The item to generate the translation for.
+     */
+    private void itemIdTranslation(Supplier<? extends Item> item) {
+        addItem(item, idToTranslation(item.get().getRegistryName().getPath()));
+    }
```

Comment:
use RegistryObject directly
```suggestion
    private void itemIdTranslation(RegistryObject<? extends Item> item) {
        addItem(item, idToTranslation(item.getId().getPath()));
    }
```